cosmos/client.go

```diff
@@ -172,3 +170,25 @@ func (c *Client) Stream(ctx context.Context, query string) (chan hash.Hash, chan
 	}()
 	return hashC, errC, nil
 }
+
+// GetAccountNumberSequenceAndIncrement returns the account number and sequence but also increment the sequence.
+func (c *Client) getAccountSequenceAndIncrement() (uint64, error) {
+	getSequenceMutex.Lock()
+	defer getSequenceMutex.Unlock()
+	accKb, err := c.kb.Get(c.accName)
+	if err != nil {
+		return 0, err
+	}
+	// get local seq
+	seq := sequence
+	// get remote seq
+	if acc, err := auth.NewAccountRetriever(c).GetAccount(accKb.GetAddress()); err == nil {
+		// replace seq if sup
+		if acc.GetSequence() > seq {
+			seq = acc.GetSequence()
+		}
+	}
+	// increment local sequence
+	sequence = seq + 1
+	return seq, nil
```

Comment:
s/ 	return seq, nil / 	return sequence, nil /

As we return greather value ?

Reply:
Also this function can be simplified a bit like:

```go
func (c *Client) getAccountSequenceAndIncrement() (uint64, error) {
  accKb, err := c.kb.Get(c.accName)
  if err != nil {
    return 0, err
  }
  acc, err := auth.NewAccountRetriever(c).GetAccount(accKb.GetAddress())
  if err != nil {
    return 0, err
  }

  getSequenceMutex.Lock()
  sequence = xmath.MaxUint64(acc.GetSequence(), sequence) + 1
  getSequenceMutex.Unlock()
  return sequence, nil
}
```

Reply:
> s/ return seq, nil / return sequence, nil /
> 
> As we return greather value ?

The sequence is increased for the next use. I respect the default implementation of cosmos GetSequence system. First time you get the sequence, it returns 0. Next time (after 1 block) it returns 1.

Reply:
> ```go
> xmath
> ```

`auth.NewAccountRetriever` returns an error if the account doesn't exist (and it's the case when it has not received any token yet). So it need to bypass the error.
But otherwise the implementation looks correct. I will improve mine.

Reply:
> where is xmath.MaxUint64 implemented?

It isn't but it great helper function for our `x` dir :)

> auth.NewAccountRetriever returns an error if the account doesn't exist (and it's the case when it has not received any token yet). 

Oh right, so maybe we can keep the way you wrote :)